### ISSUE TITLE
an incompatible Mode prevents the PDE from quitting after the last window is closed

### ISSUE THREAD

**Issue by mxa** [processing/processing#5112]
On Linux (last running version 3.3) closing the last processing window doesn't quit processing, but leaves the process running without a possibility to reach it. <kbd>ctl</kbd>+<kbd>c</kbd> in the terminal window where processing has been started from will quit it.

Possible solutions for this issue would be:
1. End the process after the last window has been closed.
2. Use a tabbed window instead of opening each sketch in a new environment
3. When closing last window, ask user what to do (open blank new sketch, quit...)

**Comment by GKFX** [comment]
How are you starting Processing and what sequence of events lets you see this? On my Linux machine (running the latest version from Git) the command `./processing` terminates immediately, so <kbd>ctrl</kbd>+<kbd>c</kbd> does nothing. On closing the final window I see the java process terminate as expected.

**Comment by mxa** [comment]
In the /processing-3.3 i run processing with `./processing` then if I close the window, the process keeps running without any windows open.

**Comment by GKFX** [comment]
Weird. I've tried the official download and I'm afraid I still can't see it.

**Comment by mxa** [comment]
Same is the case for processing 3.3.4 which I can now run by replacing the broken bundled java with the system one.

**Comment by mxa** [comment]
Issue is still there in processing 3.3.5.

```
JavaScriptMode is not compatible with this version of Processing
ExperimentalMode is not compatible with this version of Processing
TweakMode is not compatible with this version of Processing
```

closing the the last window doesn't bring me back to the command prompt, I have to press Ctl.+c to really quit it.

**Comment by benfry** [comment]
I can't reproduce this. 

Are you using an incompatible version of Java, rather than the one included with the download?

**Comment by mxa** [comment]
just tested with processing 3.3.6 same there.

Steps to reproduce:
1. Download processing
2. extract
3. do`./processing`
4. terminal shows  `JavaScriptMode is not compatible with this version of Processing
ExperimentalMode is not compatible with this version of Processing
TweakMode is not compatible with this version of Processing`
5. close processing window
6. look in terminal, the process is still running.

**Comment by benfry** [comment]
While Processing is not running, can you try renaming your sketchbook so that next time you run, it creates a new sketchbook folder? It looks like you have a handful of Modes and/or Libraries installed, and I wonder if one of them is causing the problem.

**Comment by mxa** [comment]
@benfry indeed, when the sketchbook folder is renamed an a new one is created the terminal gets detached immediately after typing `./processing`. Interesting.

**Comment by benfry** [comment]
Ok, if you'd like to give us a hand, you could also try re-adding various items to your sketchbook and see which one is making the PDE hang. i.e. you could move the old `modes` folder back and see if the problem comes back, then do the modes one-by-one, etc…

**Comment by mxa** [comment]
@benfry Sure I'd like to give a hand. It's any of the three modes that throw a line to the console which make Processing not quit.

**Comment by benfry** [comment]
Ok, let me know if you can figure out which one. (Or perhaps it's any of them—that any Mode that doesn't load properly causes the trouble.)

**Comment by mxa** [comment]
Sorry to be unclear. Each of the three libraries alone causes Processing to not quit.

**Comment by mxa** [comment]
It may also be that posting that error line to the terminal is the cause for not quitting. (Just a guess.)

**Comment by benfry** [comment]
Ok, thanks. Updated the title and re-opening so it can get fixed.
